tachyon/crypto/commitments/kzg/kzg.h

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_CRYPTO_COMMITMENTS_KZG_KZG_H_
+#define TACHYON_CRYPTO_COMMITMENTS_KZG_KZG_H_
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/math/elliptic_curves/msm/variable_base_msm.h"
+#include "tachyon/math/elliptic_curves/point_conversions.h"
+#include "tachyon/math/polynomials/univariate/radix2_evaluation_domain.h"
+#include "tachyon/math/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon::crypto {
+
+template <typename G1PointTy, typename G2PointTy>
+class KZGParams {
+ public:
+  using Fr = typename G1PointTy::ScalarField;
+  using Bucket = typename math::Pippenger<G1PointTy>::Bucket;
+  using UnivariateEvaluationDomainTy = math::Radix2EvaluationDomain<Fr>;
```

Comment:
We decided to create the evaluation domain through a factory. See, univariate_evaluation_domain_factory.h. And the factory creates the `UnivariateEvaluationDomain` (not `Radix2EvaluationDomain`).